src/ophyd_async/epics/areadetector/controllers/kinetix_controller.py

```diff
@@ -0,0 +1,55 @@
+import asyncio
+from typing import Optional, Set
+
+from ophyd_async.core import AsyncStatus, DetectorControl, DetectorTrigger
+from ophyd_async.epics.areadetector.drivers.ad_base import (
+    DEFAULT_GOOD_STATES,
+    DetectorState,
+    start_acquiring_driver_and_ensure_status,
+)
+
+from ..drivers.kinetix_driver import KinetixDriver, KinetixTriggerMode
+from ..utils import ImageMode, stop_busy_record
+
+KINETIX_TRIGGER_MODE_MAP = {
+    DetectorTrigger.internal: KinetixTriggerMode.internal,
+    DetectorTrigger.constant_gate: KinetixTriggerMode.gate,
+    DetectorTrigger.variable_gate: KinetixTriggerMode.gate,
+    DetectorTrigger.edge_trigger: KinetixTriggerMode.edge,
+}
+
+
+class KinetixController(DetectorControl):
+    def __init__(
+        self,
+        driver: KinetixDriver,
+        good_states: Set[DetectorState] = set(DEFAULT_GOOD_STATES),
```

Comment:
I don't know enough about AD DetectorStates, so I ignored this param for the Pilatus and Aravis- do we think it's important to expose? I can go back and adjust those detectors before there's any pickup.

Reply:
I see no reason to have to override the good states when instantiating the Controller, I think we know up front what are the good states for a given detector.

@jwlodek any reason you added it as an init arg? If not then I recommend we:

```suggestion
```

Reply:
The only reason was that it was done this way for the Pilatus when I was writing these classes.